lib/classes/Variables.js

```diff
@@ -344,11 +344,24 @@ class Variables {
     }, valueToPopulate);
   }
 
-  warnIfNotFound(variableString, valueToPopulate) {
+  deleteNullEnvironmentVariables() {
```

Comment:
Still not 100% sure if this should be kept here since this is not Serverless Variable specific (and this file only contains the logic for the Serverless Variables system).

I would see this somewhere in the `package` or `deploy` plugin (not the AWS one though).